tests/apollo/test_skvbc_preexecution.py

```diff
@@ -191,3 +192,33 @@ async def test_view_change(self, bft_network):
                 break
 
         self.assertEqual(new_last_block, last_block)
+
+    @with_trio
+    @with_bft_network(start_replica_cmd)
+    async def test_parallel_tx_after_f_nonprimary_crash(self, bft_network):
+        '''
+        Crash f nonprimary replicas and submit X parallel write submissions.
+        Block processing of the network should be unaffected with f-count interruption.
+        Final block length should match submitted transactions count exactly.
+        '''
+        bft_network.start_all_replicas()
+        skvbc = kvbc.SimpleKVBCProtocol(bft_network)
+        await trio.sleep(SKVBC_INIT_GRACE_TIME)
+        
+        read_client = bft_network.random_client()
+        submit_clients = bft_network.random_clients(MAX_CONCURRENCY)
+        num_of_requests = 10 * len(submit_clients) # each client will send 10 tx
+        nonprimaries = bft_network.all_replicas(without={0}) # primary index is 0
+        crash_targets = random.sample(nonprimaries, bft_network.config.f) # pick random f to crash
+        bft_network.stop_replicas(crash_targets) # crash chosen nonprimary replicas
+
+        await self.run_concurrent_pre_execution_requests(skvbc, submit_clients, num_of_requests, write_weight=1.0)
+        final_block_count = skvbc.parse_reply(await read_client.read(skvbc.get_last_block_req()))
+
+        print("")
+        print(f"Randomly picked replica indexes {crash_targets} (nonprimary) to be stopped.")
+        print(f"Total of {num_of_requests} write pre-exec tx, "
+              f"concurrently submitted through {len(submit_clients)} clients.")
+        print(f"Finished at block {final_block_count} (expected >= {num_of_requests}).")
+        self.assertTrue(final_block_count >= num_of_requests / len(submit_clients))
```

Comment:
I've checked again in the code and discussed with Yulia. My mistake - it turns out each client request creates a separate block. So the following check should work:
`self.assertTrue(final_block_count == num_of_requests)`

Reply:
Will change to `self.assertTrue(final_block_count == num_of_requests)`.

Filing Jira bug and debug log for 3 failed runs with `client.write` hang-up.

Thanks!